Plugin/CartIntercept.php

```diff
@@ -128,11 +128,13 @@ public function afterUpdateItems(Cart $cardModel, $cart)
     public function isAnonymousReady($storeId = null)
     {
         if ($this->sailthruSettings->canAbandonAnonymous($storeId) && $hid = $this->sailthruCookie->get()) {
-            $response = $this->clientManager->getClient($storeId)->getUserByKey($hid, 'cookie', ['keys' => 1]);
-            if (array_key_exists('keys', $response)) {
-                $email = $response['keys']['email'];
+            if ($hid != 'do-not-track') {
```

Comment:
Why dont we add it as part of above if i.e. just change
if ($this->sailthruSettings->canAbandonAnonymous($storeId) && $hid = $this->sailthruCookie->get()) {
to (pls. verify syntax)
if ($this->sailthruSettings->canAbandonAnonymous($storeId) && ($hid != 'do-not-track' && $hid = $this->sailthruCookie->get()) ) {

Reply:
Yes right, will combine this.

Reply:
Actually it is showing error: undefined variable $hid in that case because setting the value for $hid using `$hid = $this->sailthruCookie->get()` so i think there are two ways either move `$hid = $this->sailthruCookie->get()` before if statement and then check and verify hid value or keep as is in second if.

Reply:
```
$hid = $this->sailthruCookie->get();
        if ($this->sailthruSettings->canAbandonAnonymous($storeId) && $hid != null && $hid != '' && $hid != 'do-not-track')
```

Reply:
Like above we can add..

Reply:
or we can just return as a first statement inside if i.e if($hid == 'do-not-track') return; we can also add null or empty check (in all 3 scenarios we can return without processing further)